src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -24,6 +24,28 @@ public void MultipleProcesses_StartAllKillAllWaitAll()
             foreach (Process p in processes) Assert.True(p.WaitForExit(WaitInMS));
         }
 
+        [Fact]
+        public async Task MultipleProcesses_StartAllKillAllWaitAllAsync()
+        {
+            const int Iters = 10;
+            Process[] processes = Enumerable.Range(0, Iters).Select(_ => CreateProcessLong()).ToArray();
+
+            foreach (Process p in processes) p.EnableRaisingEvents = true;
+            foreach (Process p in processes) p.Start();
+            foreach (Process p in processes) p.Kill();
+            foreach (Process p in processes)
+            {
+                using (var cts = new CancellationTokenSource(WaitInMS))
+                {
+                    var task = p.WaitForExitAsync(cts.Token);
+                    await task;
+                    Assert.True(task.IsCompletedSuccessfully);
+                    Assert.False(task.IsCanceled);
```

Comment:
Is this necessary? Seems like you're testing that `await` functions correctly here, not anything specific to `WaitForExitAsync`.

Reply:
Since I was working with a `TaskCompletionSource` directly I figured it was better check these properties as well.

If you prefer I remove them let me know and I can do that.

Reply:
I'd prefer them removed; avoids some noise in the tests.

Reply:
Removed!